### ISSUE TITLE
MacOS system Ruby version (2.6) being used and prevents the extension from working

### ISSUE THREAD

**Issue by pprotas** [Shopify/ruby-lsp#2921]
#### VS Code Version

1.95.3

#### Ruby LSP Extension Version

0.8.14

#### Ruby LSP Server Version

0.21.3

#### Ruby LSP Addons

- Ruby LSP RSpec
- Reek: Code smell detector for Ruby
- Ruby LSP Rails

#### Ruby Version

3.2.5

#### Ruby Version Manager

mise

#### Installed Extensions

<details>
<summary>Click to expand</summary>

- auto-rename-tag (0.1.10)
- copilot (1.246.0)
- copilot-chat (0.22.4)
- debugpy (2024.12.0)
- gitlab-workflow (5.19.0)
- gitlens (16.0.4)
- prettier-vscode (11.0.0)
- python (2024.20.0)
- remote-containers (0.391.0)
- ruby-lsp (0.8.14)
- sqltools (0.28.3)
- sqltools-driver-pg (0.5.4)
- vim (1.28.1)
- vscode-docker (1.29.3)
- vscode-eslint (3.0.10)
- vscode-html-css (2.0.11)
- vscode-pull-request-github (0.101.2024111810)
- vscode-pylance (2024.11.3)
- vscode-tailwindcss (0.12.15)
</details>

#### Ruby LSP Settings

<details>
<summary>Click to expand</summary>

##### Workspace

```json
{}
```

##### User

```json
{
  "enabledFeatures": {
    "codeActions": true,
    "diagnostics": true,
    "documentHighlights": true,
    "documentLink": true,
    "documentSymbols": true,
    "foldingRanges": true,
    "formatting": true,
    "hover": true,
    "inlayHint": true,
    "onTypeFormatting": true,
    "selectionRanges": true,
    "semanticHighlighting": true,
    "completion": true,
    "codeLens": true,
    "definition": true,
    "workspaceSymbol": true,
    "signatureHelp": true,
    "typeHierarchy": true
  },
  "featuresConfiguration": {},
  "addonSettings": {},
  "rubyVersionManager": {
    "identifier": "mise"
  },
  "customRubyCommand": "",
  "formatter": "auto",
  "linters": null,
  "bundleGemfile": "",
  "testTimeout": 30,
  "branch": "",
  "pullDiagnosticsOn": "both",
  "useBundlerCompose": false,
  "bypassTypechecker": false,
  "rubyExecutablePath": "",
  "indexing": {},
  "erbSupport": true,
  "useLauncher": false,
  "featureFlags": {}
}
```

</details>

### Reproduction steps

1. Open VSC on MacOS using Finder/Spotlight with the Ruby LSP extension
2. Open a Ruby file
3. Notice it does not work and tries to use the MacOS 2.6 ruby version:
```
Failed to setup the bundle: Command failed: gem list ruby-lsp language_server-protocol prism rbs sorbet-runtime /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby: invalid option --yjit (-h will show valid options) (RuntimeError) . See [Troubleshooting](https://shopify.github.io/ruby-lsp/troubleshooting.html) for help
```
5. Run VSC on MacOS using `code .` from your terminal
6. Notice it works fine and doesn't try to use the 2.6 ruby version, but correctly uses 3.2.5 from Mise version manager

Basically it looks like `asyncExec` in `vscode/src/workspace.ts:271` uses the MacOS built-in 2.6 Ruby version by default to run `gem list`. But when I run VSC from my terminal, it correctly loads the 3.2.5 Ruby version from Mise.

**Comment by pprotas** [comment]
Yes it was, I will try your suggestion

**Comment by pprotas** [comment]
Just wanted to comment that I found the cause of this issue. VSCode was not using the Ruby version provided by Mise, unless you launched it with `code .` in the terminal. The fix is to download the `mise` [VSC extension:](https://marketplace.visualstudio.com/items?itemName=hverlin.mise-vscode), which makes sure VSCode uses the correct tool versions.

Fix found [here](https://mise.jdx.dev/ide-integration.html), under the section "IDE Plugins". I imagine that adding the mise shims to PATH would also fix the issue.

**Comment by vinistock** [comment]
There's something else missing from our Mise integration then. The right fix should not be to install an extra extension as we do provide Mise integration.

If you can reproduce this consistently and are willing to help debug it, you can follow the instructions on [contributing](https://shopify.github.io/ruby-lsp/contributing.html) to debug the VS Code extension locally and put a breakpoint [here](https://github.com/Shopify/ruby-lsp/blob/d4f79820f45d28baa0f3bee62b89396ec5bf7b4e/vscode/src/ruby/mise.ts#L13).

Then check what's happening each step of the way and what exactly is failing. As long as we find the Mise executable and can successfully executed `mise x -- ruby`, the integration should be working and should not require launching from the terminal.

Do you have any customization in shell scripts? Note that the NodeJS process executing VS Code extensions has no knowledge of anything configured in shell scripts. For example, if you do something like `RUBY_VERSION=3.3.1` to set a global default, that environment variable does not exist in the NodeJS process running the extension.
